nameko_statsd/statsd_dep.py

```diff
@@ -29,12 +30,21 @@ def __getattr__(self, name):
         if name in ('incr', 'decr', 'gauge', 'set', 'timing'):
             return partial(self._passthrough, name)
         else:
-            return super(LazyClient, self).__getattr__(name)
+            message = "'{cls}' object has no attribute '{attr}'".format(
+                cls=self.__class__.__name__, attr=name
+            )
+            raise AttributeError(message)
 
     def _passthrough(self, name, *args, **kwargs):
         if self.enabled:
             return getattr(self.client, name)(*args, **kwargs)
 
+    def timer(self, *args, **kwargs):
+        if self.enabled:
+            return self.client.timer(*args, **kwargs)
+        else:
+            return MagicMock()
```

Comment:
Just to check, I can't imagine what was wrong with

```python
return contextmanager(lambda *a, **k: iter([...]))
```

Reply:
That would work when using `timer` as a context manager, but not when using `timer` directly (see https://statsd.readthedocs.io/en/latest/timing.html#using-a-timer-object-directly).